addons/mail/static/src/models/file_uploader.js

```diff
@@ -148,6 +148,9 @@ registerModel({
                     }
                 }
             }
+            for (const i in attachments) {
+                (composer || thread).messaging.models['Attachment'].insert(attachments[i]);
+            }
```

Comment:
By inserting at the very end, this means visually that when 10 attachments out of 20 have been uploaded, it visually only display 10 uploading. The 10 that have finish upload are not on UI unless all attachments have been uploaded.

This is not great. And this would also be a lie to remove `uploadingAttachment.delete()`, though this is much minor than just showing nothing in the meantime.

I haven't checked yet why the `Attachment.insert()` in the `try/catch` prevent uploading most of the file. Will check now